### PR TITLE
apply codemod: assign-component-to-displayname-variable to client/ folder

### PR DESCRIPTION
**Pull request by samouri**
new codemod `assign-component-to-displayname-variable` was introduced in: https://github.com/Automattic/wp-calypso/pull/18710

This pr was generated by running:
```
1. /bin/codemods/run.js assign-component-to-displayname-variable client
2. commit
3. npm run reformat-files
4. commit
```

Since this PR changes the indentation of every file is touches, its best to view diffs with whitespace turned off by adding a `?w=1` to the end of the gh diff viewer url. [like this](https://github.com/Automattic/wp-calypso/pull/18878/files?w=1)

It should help pave the way for [running](https://github.com/Automattic/wp-calypso/pull/18889) the `react-create-class` codemod

### CODE REVIEW

File: bin/codemods/src/assign-component-to-displayname-variable.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ export default function transformer( file, api ) {
 
 	// noop if the file does not have a default export of a react component that has a displayName
 	if ( ! defaultExportDeclaration || ! component || ! displayName ) {
-		return file;
+		return;
```
Comment (spen) [inline bin/codemods/src/assign-component-to-displayname-variable.js]: The empty `Return;` seems to have done the trick :)

**Comment by spen** [comment]
Quite a shame that the diff adds/replaces near enough the whole file rather than the wrapper only - makes it easier for any would-be issues to get missed :/

**Comment by samouri** [comment]
review this with: https://github.com/Automattic/wp-calypso/pull/18878/files?w=1

diff is down to 400 lines

File: client/me/next-steps/index.jsx [hunk 2]
```diff
@@ -21,200 +21,198 @@ import observe from 'lib/mixins/data-observe';
 import sitesFactory from 'lib/sites-list';
 const sites = sitesFactory();
 
-/* eslint-enable no-restricted-imports */
```
Comment (samouri) [inline client/me/next-steps/index.jsx]: here*

**Comment by ockham** [comment]
The output looks really way nice. Great job @spen! 😄 

Looks like this needs a rebase/regenerate. After that, I guess it should get the usual e2e tests/merge/let sit in staging for a while treatment.

**Comment by ockham** [comment]
(Rebase might be sufficient in this case.)

**Comment by ockham** [comment]
> (Rebase might be sufficient in this case.)

Oh well, just had a look at the conflict 😄 Maybe regenerating is easier...

**Comment by bsessions85** [comment]
E2E tests all passed.

**Comment by ockham** [comment]
The changes to https://github.com/Automattic/wp-calypso/pull/18878/files?w=1#diff-ccb493b2e6a2212117702fc13d25327a might be controversial -- do we want to revert them?

**Comment by ockham** [comment]
The `/* eslint-enable no-restricted-imports */` comment in `client/me/next-steps/index.jsx` was lost again.

**Comment by samouri** [comment]
@ockham: `action-types` get prettified just like anything else. shame on us if we can't keep the names at reasonable lengths

**Comment by ockham** [comment]
> @ockham: `action-types` get prettified just like anything else. shame on us if we can't keep the names at reasonable lengths

Fair nuff 😄 I guess I was saying we _could_ consider changing the max char count per line, but okay :man_shrugging:

**Comment by samouri** [comment]
> Fair nuff 😄 I guess I was saying we could consider changing the max char count per line, but okay 🤷‍♂️

@ockham, I don't have any strong preference on line lengths! I'd be more than happy to approve a pr increasing/decreasing it.  The only thing i feel strongly about is that no human is deciding the style on a case by case basis :) 


That said, I do think the action types in question seem pretty...unnecessarily long

**Comment by spen** [comment]
No conflicts! 🎉 Good work @samouri!

**Pull request merged**